tests/load/test_event_stream_perf.py

```diff
@@ -0,0 +1,108 @@
+import os
+import tempfile
+import time
+import uuid
+from dataclasses import dataclass
+from typing import List
+
+import pytest
+
+from openhands.events.event import Event, EventSource
+from openhands.events.stream import EventStream
+from openhands.storage.local import LocalFileStore
+
+
+@dataclass
+class TestEvent(Event):
+    """Simple event class for testing."""
+
+    action: str = 'message'  # Makes it an action event
+    message: str = ''
+    content: str = ''  # Required for message actions
+
+
+def create_test_events(count: int) -> List[TestEvent]:
+    """Create a list of test events."""
+    return [
+        TestEvent(
+            action='message', message=f'Test message {i}', content=f'Test content {i}'
+        )
+        for i in range(count)
+    ]
+
+
+class TestEventStreamPerformance:
+    @pytest.fixture
+    def temp_dir(self):
+        """Create a temporary directory for the test."""
+        with tempfile.TemporaryDirectory() as temp_dir:
+            yield temp_dir
+
+    @pytest.fixture
+    def event_stream(self, temp_dir):
+        """Create an event stream with a local file store."""
+        file_store = LocalFileStore(temp_dir)
+        stream = EventStream(sid=str(uuid.uuid4()), file_store=file_store)
+        return stream
+
+    def test_event_stream_performance(self, event_stream):
+        """Test performance of event stream operations."""
+        NUM_EVENTS = 150
```

Comment:
Well, I had event streams of 2k events... 😅 

We may want to stress test this more.